packages/webpack-extensive-lodash-replacement-plugin/index.js

```diff
@@ -332,74 +330,106 @@ const LODASH_MODULE_NAMES = [
 		}
 
 		try {
-			const packageJson = JSON.parse( data );
-			this.lodashVersion =
-				packageJson && packageJson.dependencies && packageJson.dependencies[ 'lodash-es' ];
+			const baseLodashES = await getModuleForPath( this.moduleResolver, this.baseDir, 'lodash-es' );
+			this.baseLodashESVersion =
+				baseLodashES &&
+				baseLodashES.descriptionFileData &&
+				baseLodashES.descriptionFileData.version;
 		} catch ( error ) {
-			throwError( 'Could not parse root package.json.', error );
-		}
-
-		if ( ! this.lodashVersion ) {
-			throwError( 'No lodash-es dependency in root package.json.' );
+			throw createError( 'Could not find root `lodash-es`.' );
 		}
 
-		if ( ! semver.valid( this.lodashVersion ) ) {
-			throwError( `Invalid root package.json lodash-es version: ${ this.lodashVersion }.` );
+		if ( ! this.baseLodashESVersion ) {
+			throw createError( 'Could not determine root `lodash-es` version.' );
 		}
 	}
 
-	// Figure out the requested Lodash version range for a given import.
-	// It looks at the file with the import and traverses upwards, until it finds a package.json file
-	// with the requested dependency. It returns this range.
-	findRequestedLodashRange( file, packageName ) {
-		const finder = findPackageJson( path.dirname( file ) );
-		let version;
-
-		try {
-			while ( ! version ) {
-				const found = finder.next();
-
-				if ( found.filename === this.baseFile ) {
-					return this.lodashVersion;
-				}
+	// Figure out the version for a given import.
+	// It follows the node resolution algorithm until it finds the package, returning its version.
+	async findRequestedVersion( file, packageName ) {
+		const foundContext = await getModuleForPath(
+			this.moduleResolver,
+			path.dirname( file ),
+			packageName
+		);
+		const baseLodashContext = this.baseLodashContext;
 
-				const contents = found && found.value;
-				version = contents && contents.dependencies && contents.dependencies[ packageName ];
-			}
-		} catch ( error ) {
-			throwError( `Could not find requested range for import on ${ file }.`, error );
+		if ( foundContext.path === baseLodashContext.path ) {
+			return this.baseLodashESVersion;
 		}
 
-		return version;
+		return (
+			foundContext && foundContext.descriptionFileData && foundContext.descriptionFileData.version
+		);
 	}
 
 	// Figure out if the requested Lodash import can be replaced with global lodash-es.
-	// It takes the importer's lodash version range and the global lodash-es version into account.
-	canBeReplaced( file, packageName ) {
-		const requestedVersion = this.findRequestedLodashRange( file, packageName );
-		return requestedVersion && semver.satisfies( this.lodashVersion, requestedVersion );
+	// It takes the importer's version and the global lodash-es version into account.
+	async canBeReplaced( file, packageName ) {
+		const importVersion = await this.findRequestedVersion( file, packageName );
+		const isVersionMatch =
+			importVersion &&
+			this.baseLodashESVersion &&
```

Comment:
Thanks to the early error exit during init, `this.baseLodashESVersion` is always valid here.

Reply:
Removed the check!